src/ol/style/icon.js

```diff
@@ -172,6 +174,47 @@ ol.inherits(ol.style.Icon, ol.style.Image);
 
 
 /**
+ * Clones the style.
+ * @return {ol.style.Icon} The cloned style.
+ * @api
+ */
+ol.style.Icon.prototype.clone = function() {
+  var oldImage = this.getImage(1);
+  var newImage;
+  if (this.iconImage_.getImageState() === ol.Image.State.LOADED) {
+    if (oldImage.tagName.toUpperCase() === 'IMG') {
+      newImage = /** @type {Image} */ (this.getImage(1).cloneNode(true));
```

Comment:
With reusing I meant replacing `this.getImage(1)` with the already existing `oldImage` so that the line reads:

```js
newImage = /** @type {Image} */ (newImage.cloneNode(true));
```